pkg/frontend/session.go

```diff
@@ -395,6 +396,7 @@ func (bgs *BackgroundSession) Close() {
 		bgs.Session.txnCompileCtx = nil
 		bgs.Session.txnHandler = nil
 		bgs.Session.gSysVars = nil
+		bgs.Session.statsCache = nil
```

Comment:
This won't fix memory leak.   Still, at most one statsCache per session.   It may release, but our problem is that stats cache reach GB of memories, but really, it should be at most a few MB if not KBs.

Reply:
thats because there are  lots of BackgroundSession. and backgroudsession dont need stats cache